### ISSUE TITLE
setTimeout(...).unref is not a function

### ISSUE THREAD

**Issue by yafp** [microlinkhq/metascraper#267]
### Prerequisites

- [ x ] I'm using the last version.
- [ x ] My node version is the same as declared as `package.json`.

### Subject of the issue
I am using metascraper since a while in my project and it worked great. Recently i've seen updates on both sides
* metascraper and
* got

today i updated both in my dev environment and got the following error when trying to fetch meta data

```
RequestError: setTimeout(...).unref is not a function
    at ClientRequest.<anonymous> (/home/x/y/node_modules/got/dist/source/core/index.js:786:25)
    at Object.onceWrapper (events.js:300:26)
    at ClientRequest.emit (events.js:215:7)
    at ClientRequest.EventEmitter.emit (domain.js:476:20)
    at ClientRequest.origin.emit (/home/x/y/node_modules/@szmarczak/http-timer/dist/source/index.js:39:20)
    at TLSSocket.socketErrorListener (_http_client.js:406:9)
    at TLSSocket.emit (events.js:210:5)
    at TLSSocket.EventEmitter.emit (domain.js:476:20)
    at emitErrorNT (internal/streams/destroy.js:92:8)
    at emitErrorAndCloseNT (internal/streams/destroy.js:60:3)
    at processTicksAndRejections (internal/process/task_queues.js:80:21)
    at CacheableLookup.tick (/home/x/y/node_modules/cacheable-lookup/source/index.js:290:22)
    at CacheableLookup.query (/home/x/y/node_modules/cacheable-lookup/source/index.js:176:8)
    at CacheableLookup.lookupAsync (/home/x/y/node_modules/cacheable-lookup/source/index.js:137:27)
    at CacheableLookup.lookup (/home/x/y/node_modules/cacheable-lookup/source/index.js:121:8)
    at net.js:1030:5
    at defaultTriggerAsyncIdScope (internal/async_hooks.js:301:12)
    at lookupAndConnect (net.js:1029:3)
    at TLSSocket.Socket.connect (net.js:963:5)
    at Object.connect (_tls_wrap.js:1434:13)
    at Agent.createConnection (https.js:125:22)
    at Agent.createSocket (_http_agent.js:234:26)
    at Agent.addRequest (_http_agent.js:193:10)
    at new ClientRequest (_http_client.js:276:16)
    at request (https.js:309:10)
    at PromisableRequest._makeRequest (/home/x/y/node_modules/got/dist/source/core/index.js:904:39)
    at /home/x/y/node_modules/got/dist/source/core/index.js:265:28

```

Is that a known issue?

my function is pretty much based on the example:

```

    const metascraper = require('metascraper')([
        require('metascraper-video')(),
        require('metascraper-description')(),
        require('metascraper-image')(),
        require('metascraper-audio')(),
        require('metascraper-logo')(),
        require('metascraper-logo-favicon')(),
        require('metascraper-media-provider')(),
        require('metascraper-soundcloud')(),
        require('metascraper-title')(),
        require('metascraper-youtube')()
    ])

    var urlLogo
    var urlImage
    var urlDescription
    var urlTitle
    var urlAudio

    const got = require('got')

    const targetUrl = url

    ;(async () => {
        const { body: html, url } = await got(targetUrl)
        const metadata = await metascraper({ html, url })

        urlLogo = metadata.logo
        urlImage = metadata.image
        urlDescription = metadata.description
        urlTitle = metadata.title
        urlAudio = metadata.audio
    })()

```

### Steps to reproduce
Unsure so far.

### Expected behaviour
Should give me metadata

### Actual behaviour
I seem unable to fetch metadata

**Comment by Kikobeats** [comment]
Hello,

I recently noted got had some regressions; I updated it to the latest version that fixes this kind of problem.

Can you try `metascraper@5.11.11` version?

**Comment by yafp** [comment]
Hey,

its the first time that metascraper and its addons seem to have different version.
According to npm-check i am up-to-date with all my packages.

```
"metascraper": "^5.11.10",
    "metascraper-audio": "^5.11.10",
    "metascraper-description": "^5.11.10",
    "metascraper-image": "^5.11.10",
    "metascraper-logo": "^5.11.10",
    "metascraper-logo-favicon": "^5.11.11",
    "metascraper-media-provider": "^5.11.11",
    "metascraper-soundcloud": "^5.11.10",
    "metascraper-title": "^5.11.10",
    "metascraper-video": "^5.11.10",
    "metascraper-youtube": "^5.11.10",
```

so only
* metascraper-logo-favicon, and 
* metascraper-media-provider

are provided in 5.11.11.

**Comment by Kikobeats** [comment]
hm, can you make the error reproducible?

e.g., remove unnecessary plugins, try to isolate the plugin that is causing the error.

Also, try different urls to input to determinate if the url caused the error

**Comment by yafp** [comment]
so far i got the impression this error happens on all urls i am forwarding to metascaper ... which is strange.

gonna try to isolate the issue

**Comment by yafp** [comment]
As a first step i moved back to the old versions - as in:
* got 10.7.0
and
* metascraper 5.11.8

As that was the last working setup i remember. and yes- using those versions my project is working without issues. gonna see if i can find more details ..

**Comment by yafp** [comment]
Alright it seems unrelated with the url. Tried the following combinations: 

* metascraper 5.11.8 & got 10.7.0 is working
* metascraper 5.11.10/11 and got 10.7.0 is working
* metascraper 5.11.8 & got 11.0.3 is not working -> causing the error
* metascraper 5.11.10/11 and got 11.0.3 is not working -> causing the error

**Comment by Kikobeats** [comment]
then is clearly a `got` thing related to 11.x version.

I suggest you open an issue at [got](http://ghub.io/got) repo, I can't see anything I can do from my side :(

**Comment by yafp** [comment]
yes - think so too.

**Comment by yafp** [comment]
just as a sidenote: it seems to be fixed with got 11.1.1

**Comment by Kikobeats** [comment]
good catch! in this case, the codebase is already updated
https://github.com/microlinkhq/metascraper/commit/620297b2dfac5a9de2306d1798336f838cb9abc0
